### PR TITLE
Feat/info api

### PR DESCRIPTION
**Pull request by Gozala**
Builds on top of #28 adds other API method

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,6 +1,78 @@
...
+      `cluster id response has failure: ${JSON.stringify(result, null, 2)}`
+    )
+  }
+
+  const {
+    id,
+    addresses,
+    version,
+    commit,
+    peername,
```
Comment (alanshaw) [inline src/index.js:56]:
```suggestion
    peerName,
```

For consistency, it's mapped like this here already: https://github.com/nftstorage/ipfs-cluster/blob/824f8425247eb3707d8b9174058cd93dda79dfe5/src/index.js#L141

**Review by alanshaw — CHANGES_REQUESTED** [review]
Code looks good, just one name mapping request.

**Review by hugomrdias — APPROVED** [review]
LGTM

File: src/index.js (hunk 1 above)
Comment (Gozala) [reply src/index.js:56]:
```suggestion
    peername: peerName,
```

File: src/interface.ts [hunk 2]
```diff
@@ -216,3 +216,18 @@ export enum TrackerStatus {
...
+
+export interface PeerInfo {
+  id: string
+  addresses: string[]
+}
+
+export interface ClusterInfo extends PeerInfo {
+  version: string
+  commit: string
+  peername: string
```
Comment (Gozala) [inline src/interface.ts:228]:
```suggestion
  peerName: string
```

File: test/all.spec.js [hunk 3]
```diff
@@ -229,9 +259,76 @@ describe('cluster.unpin', () => {
...
+
+describe('cluster.info', () => {
+  /**
+   * @param {cluster.API.ClusterInfo} info
+   */
+  const assertInfo = (info) => {
+    assertField(info, 'id')
+    assertField(info, 'version')
+    assert.equal(typeof info.commit, 'string')
+    assertField(info, 'peername')
```
Comment (Gozala) [inline test/all.spec.js:297]:
```suggestion
    assertField(info, 'peerName')
```

File: src/index.js [hunk 4]
```diff
@@ -1,6 +1,78 @@
...
+    cluster_peers_addresses: clusterPeersAddresses,
+    ipfs
+  } = result
+
+  return {
+    id,
+    addresses,
+    version,
+    commit,
+    peername,
```
Comment (alanshaw) [inline src/index.js:68]:
```suggestion
    peerName,
```

**Review by alanshaw — APPROVED** [review]
Last change should fix the build - otherwise LGTM.

**Pull request merged**